src/SocialAuthUserManager.php

```diff
@@ -139,33 +148,64 @@ public function setSessionKeysToNullify(array $session_keys) {
   /**
    * Creates and/or authenticates an user.
    *
-   * @param string $email
-   *   The user's email address.
    * @param string $name
    *   The user's name.
-   * @param string $id
-   *   The user's id in provider.
+   * @param string $email
+   *   The user's email address.
+   * @param string $pluginId
+   *   The social implementer identifier.
+   * @param string $provider_user_id
+   *   The unique id returned by the user.
    * @param string|bool $picture_url
    *   The user's picture.
    *
    * @return \Symfony\Component\HttpFoundation\RedirectResponse
    *   A redirect response.
    */
-  public function authenticateUser($email, $name, $id = NULL, $picture_url = FALSE) {
-    // Tries to load the user by their email.
-    $drupal_user = $this->loadUserByProperty('mail', $email);
-    // If user email has already an account in the site.
-    if ($drupal_user) {
-      // Authenticates and redirect existing user.
-      return $this->authenticateExistingUser($drupal_user);
+  public function authenticateUser($name, $email, $pluginId, $provider_user_id, $picture_url = FALSE) {
+
+    // Checks for record in social _auth entity.
+    $user_exist = $this->checkIfUserExists($pluginId, $provider_user_id);
+
+    // Checks if user has authenticated role and no record exist.
+    if ($this->currentUser->isAuthenticated() && !$user_exist) {
+      return $this->addUserRecord($this->currentUser->id(), $pluginId, $provider_user_id);
```

Comment:
Why are you returning the value from addUserRecord()? The method returns boolean